src/site/topics/testing/unit-test-collections.rst

```diff
@@ -158,6 +158,206 @@ Common Setup Code
+    :emphasize-lines: 2,7,47,49
+
+    private ArrayStack<Integer> classUnderTest;
+    private ArrayStack<Integer> preState;
+
+    @BeforeEach
+    void createStack() {
+        classUnderTest = new ArrayStack<>();
+        preState = new ArrayStack<>();
+    }
+
+    // Empty tests excluded here
+
+    @Test
+    void push_singleton_returnsTrue() {
+        classUnderTest.push(10);
+        assertTrue(classUnderTest.push(11));
+    }
+
+    @Test
+    void push_singleton_newTop() {
+        classUnderTest.push(10);
+        classUnderTest.push(11);
+        assertEquals(11, classUnderTest.peek());
+    }
+
+    @Test
+    void pop_singleton_returnsTop() {
+        classUnderTest.push(10);
+        assertEquals(10, classUnderTest.pop());
+    }
+
+    @Test
+    void pop_singleton_emptyStack() {
+        classUnderTest.push(10);
+        classUnderTest.pop();
+        assertEquals(new ArrayStack<>(), classUnderTest);
+    }
+
+    @Test
+    void peek_singleton_returnsTop() {
+        classUnderTest.push(10);
+        assertEquals(10, classUnderTest.peek());
+    }
+
+    @Test
+    void peek_singleton_unchanged() {
+        classUnderTest.push(10);
+        preState.push(10);
+        classUnderTest.peek();
+        assertEquals(preState, classUnderTest);
+    }
+
+    @Test
+    void isEmpty_singleton_returnsFalse() {
+        classUnderTest.push(10);
+        assertFalse(classUnderTest.isEmpty());
+    }
+
+    @Test
+    void size_singleton_returnsOne() {
+        classUnderTest.push(10);
+        assertEquals(1, classUnderTest.size());
+    }
+
+    @Test
+    void toString_singleton_returnsCorrectString() {
+        classUnderTest.push(10);
+        assertEquals("10, ", classUnderTest.toString());
+    }
+
+
+.. note::
+
+    When considering ``peek_singleton_returnsTop``, it may become clear that ``push_empty_newTop`` and
+    ``push_singleton_newTop`` are redundant tests since the ``peek_singleton_returnsTop`` will test ``peek`` after a
+    ``push`` has happened anyways, effectively checking that a ``push`` results in the expected top. When looking at the
+    code in the tests, it's clear that the tests are effectively identical. Thus, it is not really necessary to include
+    any of the ``push`` causing a new top test.
+
+        .. code-block:: java
+            :linenos:
+
+            @Test
+            void push_empty_newTop() {
+                classUnderTest.push(11);
+                assertEquals(11, classUnderTest.peek());
+            }
+
+            @Test
+            void peek_singleton_returnsTop() {
+                classUnderTest.push(10);
+                assertEquals(10, classUnderTest.peek());
+            }
+
+
+    This highlights the complexities caused by the *interconnectedness* of the collection's methods --- one cannot test
+    that ``push`` results in a new top without using ``peek``, and one cannot test ``peek`` without having already
+    called ``push``.
+
+
+
+Nested Classes
+--------------
```

Comment:
_Nested **Test** Classes_  -- just to be specific, nested classes are introduced with linked implementations later.